src/state.rs

```diff
@@ -2031,6 +2031,63 @@ where
         }
         path_str
     }
+    
+    /// returns a `String` containing a formatted view of the full path which led
+    /// to this point, in terms of LLVM IR instructions. If the Path includes any
+    /// hooked functions, the instructions which might be contained within those
+    /// functions are not included.
+    pub fn pretty_path_llvm_instructions(&self) -> String {
+        // to correctly print an instruction trace in the presence of function calls,
+        // we need to know which calls divert control flow to another basic block. This
+        // can be determined by finding all basic blocks in the path which do not
+        // begin at instruction 0.
+        let mut reenter_set = HashSet::new(); //Store (bb_name, instr_idx) of all calls that leave bb
+        for path_entry in self.get_path().iter() {
+            match path_entry.0.instr {
+                BBInstrIndex::Instr(idx) => {
+                    if idx != 0 {
+                        reenter_set.insert((path_entry.0.bb.name.clone(), idx - 1));
+                    }
+                }
+                BBInstrIndex::Terminator => {
+                    let num_instrs = path_entry.0.bb.instrs.len();
+                    if num_instrs > 0 {
+                        // tail call!
```

Comment:
I don't think this necessarily indicates a tail call?  for instance we could have
```
%4 = call i1 @foo(/* ... */)
br i1 %4, label x, label y
```
I don't think the logic here is wrong, just the comment might be deceiving